src/test-utils/src/evaluateGraphQLResolver.js

```diff
@@ -18,17 +20,30 @@ export default function evaluateGraphQLResolver(
   argsValue?: { [key: string]: any, ... },
   contextValue?: { [key: string]: any, ... },
 ): any {
-  const resolveFn =
-    field.resolve ||
-    function resolveMock(): string {
-      return testValue[field.name];
-    };
-  const resolvedValue = resolveFn(testValue, argsValue, contextValue, {
-    path: { key: 'mocked' },
-  });
+  const resolvedValue = getResolvedValue(field, testValue, argsValue, contextValue);
 
   if (typeof field.type.serialize === 'function') {
     return field.type.serialize(resolvedValue);
+  } else if (typeof field.type.ofType?.serialize === 'function') {
+    coerceInputValue(resolvedValue, field.type);
+    return field.type.ofType.serialize(resolvedValue);
   }
   return resolvedValue;
 }
+
+function getResolvedValue(field, testValue, argsValue, contextValue): mixed {
+  if (typeof field.resolve === 'function') {
+    return field.resolve(testValue, argsValue, contextValue, {
+      path: { key: 'mocked' },
+    });
+  }
+  return defaultFieldResolver(
+    testValue,
+    argsValue ?? {},
+    contextValue,
+    ({
+      fieldName: field.name,
+      path: { key: 'mocked' },
+    }: any),
```

Comment:
Would it be possible to sync these params with the direct `field.resolve` on line 36?

Reply:
I tried